xsd-core/src/main/java/io/ballerina/xsdtorecordconverter/Utils.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsdtorecordconverter;
+
+import io.ballerina.compiler.syntax.tree.AbstractNodeFactory;
+import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.NodeFactory;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.NodeParser;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl;
+import org.ballerinalang.formatter.core.Formatter;
+import org.ballerinalang.formatter.core.FormatterException;
+import org.ballerinalang.formatter.core.options.ForceFormattingOptions;
+import org.ballerinalang.formatter.core.options.FormattingOptions;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.CLOSE_BRACES;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.CONTENT_FIELD;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.SEMICOLON;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.STRING;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.TYPE;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.VERTICAL_BAR;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.WHITESPACE;
+
+/**
+ * This class contains util functions necessary for converting XSD to Ballerina Record types.
+ */
+public class Utils {
+    static ModulePartNode generateModulePartNode(HashMap<String, ModuleMemberDeclarationNode> nodes,
```

Comment:
Any reason to use `HashMap` (or the implementation class) rather than the `Map` interface in this code section ? 

IMO we should not depend on the implementation rather we should depend on the interface so that when we want to change the implementation in the future it would not break the code.